lib/src/main/java/org/exoplatform/social/addons/storage/RDBMSSpaceStorageImpl.java

```diff
@@ -195,14 +194,11 @@ public int getLastAccessedSpaceCount(SpaceFilter spaceFilter) throws SpaceStorag
   @Override
   public List<String> getMemberSpaceIds(String identityId, int offset, int limit) throws SpaceStorageException {
     Identity identity = identityStorage.findIdentityById(identityId);
-    List<Space> spaces = getMemberSpaces(identity.getRemoteId(), offset, limit);
+    List<Long> spaceIds = spaceMemberDAO.getSpacesIdsByUserName(identity.getRemoteId(), offset, limit);
 
     List<String> ids = new LinkedList<>();
-    for (Space space : spaces) {
-      Identity spaceIdentity = identityStorage.findIdentity(SpaceIdentityProvider.NAME, space.getPrettyName());
-      if (spaceIdentity != null) {
-        ids.add(spaceIdentity.getId());
-      }
+    for (Long spaceId : spaceIds) {
```

Comment:
Method getSpacesIdsByUserName can return null, so we have to check for null to avoid NPE.